src/backend/language.cpp

```diff
@@ -65,249 +86,130 @@ LanguageMap::LanguageMap() {
+    addLanguage({"kaa-Latn"}, QT_TRANSLATE_NOOP("QObject", "Kara-Kalpak (Latin script)"));
+    addLanguage({"kap-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Bezhta (Cyrillic script)"));
+    addLanguage({"kjh-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Khakas (Cyrillic script)"));
+    addLanguage({"kmr-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Kurdish, Northern (Cyrillic script)"));
+    addLanguage({"kmr-Latn"}, QT_TRANSLATE_NOOP("QObject", "Kurdish, Northern (Latin script)"));
+    addLanguage({"kpy-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Koryak (Cyrillic script)"));
+    addLanguage({"ky-Arab"}, QT_TRANSLATE_NOOP("QObject", "Kirghiz (Arabic script)"));
+    addLanguage({"ky-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Kirghiz (Cyrillic script)"));
+    addLanguage({"lez-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Lezghian (Cyrillic script)"));
+    // This next one is not in the iso_codes table - see https://iso639-3.sil.org/code/ncq
+    addLanguage({"ncq"}, QT_TRANSLATE_NOOP("QObject", "Katang, Northern"));
+    addLanguage({"nog-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Nogai (Cyrillic script)"));
+    addLanguage({"oss-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Ossetian (Cyrillic script)"));
+    addLanguage({"sah-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Yakut (Cyrillic script)"));
+    addLanguage({"tk-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Turkmen (Cyrillic script)"));
+    addLanguage({"tk-Latn"}, QT_TRANSLATE_NOOP("QObject", "Turkmen (Latin script)"));
+    addLanguage({"tkr-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Tsakhur (Cyrillic script)"));
+    addLanguage({"tt-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Tatar (Cyrillic script)"));
+    addLanguage({"tyv-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Tuvinian (Cyrillic script)"));
+    addLanguage({"uz-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Uzbek (Cyrillic script)"));
+    addLanguage({"uz-Latn"}, QT_TRANSLATE_NOOP("QObject", "Uzbek (Latin script)"));
+    addLanguage({"xal-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Kalmyk (Cyrillic script)"));
+    addLanguage({"xnj"}, QT_TRANSLATE_NOOP("QObject", "Ngoni"));
+    addLanguage({"yrk-Cyrl"}, QT_TRANSLATE_NOOP("QObject", "Nenets (Cyrillic script)"));
+    addLanguage({"zh-Hans"}, QT_TRANSLATE_NOOP("QObject", "Chinese, Simplified"));
+    addLanguage({"zh-Hant"}, QT_TRANSLATE_NOOP("QObject", "Chinese, Traditional"));
+}
+
+void LanguageMap::addLanguage(QStringList abbrevs, QString englishName)
+{
+    auto language =
+            std::make_shared<Language>(std::move(abbrevs),
+                                       std::move(englishName));
+    for (auto const & abbrev : language->abbrevs()) {
+        BT_ASSERT(!contains(abbrev));
+        insert(abbrev, language);
+    }
+}
+
+void LanguageMap::InitGetText(void)
+{
+#if !defined(__APPLE__)
+    localeDir = "/usr/share/iso-codes/json/";
+#else
+    char *iso_codes_version = ISO_CODES_INFO.VERSION
+    localeDir = sprintf("/usr/local/Cellar/iso-codes/%s/share/iso-codes/json/", iso_codes_version);
+#endif
```

Comment:
The `__APPLE__`-specific conditionally compiled code should probably be removed. We will probably have to add Apple-specific logic before release anyway, but this need not be part of this merge request.